tests/python/pants_test/pantsd/test_pantsd_integration.py

```diff
@@ -347,6 +347,34 @@ def test_pantsd_pid_change(self):
       # Remove the pidfile so that the teardown script doesn't try to kill process 9.
       os.unlink(pidpath)
 
+  def test_pantsd_memory_usage(self):
+    """Validates that after N runs, memory usage has increased by no more than X percent."""
+    number_of_runs = 10
+    max_memory_increase_fraction = 0.4
+    with self.pantsd_successful_run_context() as (pantsd_run, checker, workdir, config):
+      cmd = ['list', 'testprojects::']
+      self.assert_success(pantsd_run(cmd))
+      initial_memory_usage = checker.current_memory_usage()
+      for _ in range(number_of_runs):
+        self.assert_success(pantsd_run(cmd))
+        checker.assert_running()
+
+      final_memory_usage = checker.current_memory_usage()
+      self.assertTrue(
+          initial_memory_usage <= final_memory_usage,
+          "Memory usage inverted unexpectedly: {} > {}".format(
+            initial_memory_usage, final_memory_usage
+          )
+        )
+
+      increase_fraction = (final_memory_usage / initial_memory_usage) - 1.0
```

Comment:
Is this a place where integer division can mean we los precision?

Reply:
Hah. Apparently this changed between python 2 and 3, and I was thinking of the 3.0 behaviour.
```
Python 2.7.10 (default, Feb 22 2019, 21:17:52)
..
>>> 3/4
0
```
Python 3.7.0 (default, Jul 23 2018, 20:22:55)
```
..
>>> 3/4
0.75
```

Will fix.

Reply:
Oh, nevermind. In this file:
```
from __future__ import absolute_import, division, print_function, unicode_literals
```